xgi/classes/hypergraph.py

```diff
@@ -98,6 +98,9 @@ def __init__(self, incoming_data=None, **attr):
         self._edge = self._hyperedge_dict_factory()
         self._edge_attr = self._hyperedge_attr_dict_factory()
 
+        self.nodes = NodeView(self)
```

Comment:
wait, what does this change?
It seems we loose the explanation from the dosctrings of the nodes/edges property we had? See lines deleted below on the left.

Reply:
After talking to @nwlandry extensively about this, we realized that 1) the views were not working, 2) we don't need the lazy initialization. This PR fixes both problems. Since the views don't need to be lazy, we can just create them once and for all in the constructor.

I have added a one-line documentation for now. My next PR is going to continue changing some things around so I'd rather wait until then to add more documentation so I don't have to write everything twice :)